### PR TITLE
feat(react/vue-searchbox): add suggestion type query support

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** Feature Integration

**Description** Integration of new type of query `suggestion` to fetch suggestions of different types using a single API call.

- [x]  When we're making the suggestion query (whenever autosuggest is set to `true`, use the new suggestion type instead of the current manual calls for recent/popular suggestions.
- [x]  Remove manual API calls to `popular` suggestions and `recent` suggestions
- [x]  Update the popular/recent suggestions extraction logic from the suggestion API response
- [x] You don't need to display suggestions based on the value. The API would handle it. Currently, we display suggestions based on the value. For e.g, display recent suggestions when value is empty or popular and index suggestions when value is present.
- [x] Remove all the old prop names for supporting popular and recent suggestions.
- [x] Update docs.

[🎥  Loom Demo - 12-OCT-2021-02:54-AM](https://www.loom.com/share/8bfccff9100747e8a4fcf49c609b4177)
[🎥  Loom Demo - 13-OCT-2021-04:30-PM](https://www.loom.com/share/5585d09e362841cb908c6683e27607ab)

[📄  Docs PR](https://github.com/appbaseio/Docs/pull/241) 

[📔  Notion ](https://www.notion.so/appbase/FE-Libraries-Integration-for-suggestion-type-cba01831fea4460d9639dd3df41043b0)

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 
Functionality issues
1. Ranking is managed by BE so FE don't need any logic to display popular suggestions at top or bottom. Just render the list and change icon as per the type of suggestion.
alt de
2. The no. of popular suggestions high because of BE issue.
alt w

File: packages/vue-searchbox/src/utils/helper.js [hunk 1]
```diff
@@ -226,3 +226,18 @@ export function extractModifierKeysFromFocusShortcuts(focusShortcutsArray) {
...
+	Term: 'term',
+	Geo: 'geo',
+	Range: 'range',
+	Suggestion: 'suggestion'
+};
+
+export const suggestionTypes = {
+	Popular: 'popular',
+	Index: 'index',
+	Recent: 'recent'
```
Comment (bietkul) [inline packages/vue-searchbox/src/utils/helper.js]: Add one more type as `Promoted: 'promoted'`

File: packages/vue-searchbox/src/components/SearchComponent.jsx [hunk 2]
```diff
@@ -190,6 +201,12 @@ const SearchComponent = {
 			clearOnQueryChange,
 			distinctField,
 			distinctFieldConfig,
+			enableRecentSuggestions,
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchComponent.jsx:212]: @mohdashraf010897 The property name in searhcbox lib is `enableRecentSearches`. We don't want to break the existing functionality. So support both names. If `enableRecentSuggestions` is not defined then fallback to `enableRecentSearches`.

File: packages/vue-searchbox/src/components/SearchComponent.jsx [hunk 3]
```diff
@@ -61,7 +61,12 @@ const SearchComponent = {
 		distinctFieldConfig: VueTypes.object,
 		// subscribe on changes,
 		subscribeTo: VueTypes.arrayOf(VueTypes.string),
-		triggerQueryOnInit: VueTypes.bool.def(true)
+		triggerQueryOnInit: VueTypes.bool.def(true),
+		recentSuggestionsConfig: VueTypes.object,
+  		popularSuggestionsConfig: VueTypes.object,
+ 		maxPredictedWords: VueTypes.number,
+ 		urlField: VueTypes.string,
+  		rankFeature: VueTypes.object,
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchComponent.jsx]: You have missed some props. For e.g `enablePopularSuggestions`

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 4]
```diff
@@ -912,7 +899,8 @@ const SearchBoxWrapper = {
 		return (
 			<SearchComponent
 				value=""
-				triggerQueryOnInit={!!context.props.enablePopularSuggestions}
+				type={queryTypes.Suggestion}
+				triggerQueryOnInit={!!context.props.autosuggest}
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]: We only want to trigger query initially when `recentSuggestions` are enabled.

**Review by bietkul — CHANGES_REQUESTED** [review]
Remove suggestions parsing logic from searchcomponent. The `suggestions` property is no longer needed

File: packages/vue-searchbox/src/components/SearchComponent.jsx (hunk 3 above)
Comment (mohdashraf010897) [reply packages/vue-searchbox/src/components/SearchComponent.jsx]: it is already there @bietkul on Line 18-19

**Review by bietkul — APPROVED** [review]

**Pull request merged**